src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Enter value over 0');
+    }
+
+    if (a + b <= c || b + c <= a || a + c <= b) {
+      throw new Error('Enter correct values');
```

Comment:
The error message here could be more informative. Instead of saying 'Enter correct values', it would be helpful to explain that the sum of any two sides of a triangle must be greater than the third side.